jdbcSource/src/main/java/io/vantiq/extsrc/jdbcSource/JDBC.java

```diff
@@ -123,15 +123,27 @@ public int processPublish(String sqlQuery) throws VantiqSQLException {
                                 break;
                             case java.sql.Types.DATE:
                                 Date rowDate = queryResults.getDate(i);
-                                row.put(md.getColumnName(i), dfDate.format(rowDate));
+                                if (rowDate != null) {
+                                    row.put(md.getColumnName(i), dfDate.format(rowDate));
+                                } else {
+                                    row.put(md.getColumnName(i), null);
+                                }
                                 break;
                             case java.sql.Types.TIME:
                                 Time rowTime = queryResults.getTime(i);
-                                row.put(md.getColumnName(i), dfTime.format(rowTime));
+                                if (rowTime != null) {
+                                    row.put(md.getColumnName(i), dfTime.format(rowTime));
+                                } else {
+                                    row.put(md.getColumnName(i), null);
+                                }
                                 break;
                             case java.sql.Types.TIMESTAMP:
                                 Timestamp rowTimestamp = queryResults.getTimestamp(i);
-                                row.put(md.getColumnName(i), dfTimestamp.format(rowTimestamp));
+                                if (rowTimestamp != null) {
+                                    row.put(md.getColumnName(i), dfTimestamp.format(rowTimestamp));
+                                } else {
+                                    row.put(md.getColumnName(i), null);
+                                }
```

Comment:
So, suppose the date returned is invalid somehow or some other exception is thrown?  How are we preventing the overall crash (assuming that's what occurred)?  

That is, if there was a connector crash (or escaped exception that prevent us from leaving things in a sensible state), we need to ensure that we leave ourselves in a sensible state.  We may not be able to run the query in question, but we should be able to continue operation.